services/src/main/java/org/keycloak/authentication/AuthenticationProcessor.java

```diff
@@ -1141,7 +1146,15 @@ public static ClientSessionContext attachSession(AuthenticationSessionModel auth
             event.detail(Details.REMEMBER_ME, "true");
         }
 
+        final int clientSessions = userSession.getAuthenticatedClientSessions().size();
         ClientSessionContext clientSessionCtx = TokenManager.attachAuthenticationSession(session, userSession, authSession);
+        if (clientSessions == 0 && userSession.getStarted() == userSession.getLastSessionRefresh()
```

Comment:
I can see some potential for flaky tests (I can imagine that if testsuite is fast, the `userSession.getStarted() == userSession.getLastSessionRefresh()` will be true even if it is not the same request but following one. But maybe this won't be an issue... Let's go for this and we can improve if some new flaky tests happens after this PR is merged...

Reply:
The part `clientSessions == 0` solves that race. The idea is, if the client session is the first one in the user session (solves two very fast accesses in the same client) and not created now (solves the issue with a expired client session, I suppose expiration is more than 1 second).